File: my-folder/0485-max-consecutive-ones/solution.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        l = 0
        maxlen = 0
        for r in range(len(nums)):
            if nums[r] == 0:
                maxlen = max(r - l, maxlen)
                l = r + 1

        return max(maxlen, len(nums) - l)


        

File: my-folder/0485-max-consecutive-ones/solution.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int longestConsequetiveLen = 0;
        int l = 0, r = 0;
        while(l <= r && r < nums.length){
            if(nums[r] == 1){
                if(longestConsequetiveLen < r - l + 1){
                    longestConsequetiveLen = r - l + 1;
                }
                r++;
            }
            else{
                r++;
                l = r;
            }
        }
        return longestConsequetiveLen;
    }
}

// [1,1,0,1,1,1]
//        l
//            r 
